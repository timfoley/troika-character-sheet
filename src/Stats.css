.stats {
  width          : 100%;
  grid-area      : stats;
  display        : flex;
  flex-direction : row;
  flex-wrap      : wrap;
  align-items    : flex-start;
  justify-content: space-around;
}

.stats label {
  display       : flex;
  flex-direction: column;
  font-weight   : bold;
}

.stat__input {
  font-size    : 2.5rem;
  font-weight  : bold;
  width        : auto;
  margin-bottom: 0;
  padding      : .1em;
  text-align   : center;
  border       : 1px solid black;
}

.stat__input--sub {
  width     : 60px;
  border-top: none;
  font-size : 1.5rem;
}

.stat {
  width        : 80px;
  margin-bottom: 10px;
}

.sub-label {
  font-weight: initial;
  font-size  : .9rem;
  float      : right;
  text-align : right;
}