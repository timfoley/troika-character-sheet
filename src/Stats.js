import React from "react";

import "./Stats.css";

export function Stats(props) {
	const updateSkill = e =>
		props.updateCharacter("skill", parseInt(e.target.value));

	const updateMaxStamina = e =>
		props.updateCharacter("stamina", {
			...props.stamina,
			max: parseInt(e.target.value),
		});
	const updateRemainingStamina = e =>
		props.updateCharacter("stamina", {
			...props.stamina,
			remaining: parseInt(e.target.value),
		});

	const updateMaxLuck = e =>
		props.updateCharacter("luck", {
			...props.luck,
			max: parseInt(e.target.value),
		});
	const updateRemainingLuck = e =>
		props.updateCharacter("luck", {
			...props.luck,
			remaining: parseInt(e.target.value),
		});

	return (
		<div className="stats">
			<label className="stat skill">
				Skill
				<input
					className="stat__input stat__input--main"
					type="number"
					value={props.skill}
					onChange={updateSkill}
				/>
			</label>

			<div className="stat stamina">
				<label>
					Stamina
					<input
						className="stat__input stat__input--main"
						type="number"
						max={props.stamina.max}
						value={props.stamina.remaining}
						onChange={updateRemainingStamina}
					/>
				</label>
				<label className="sub-label">
					<input
						className="stat__input stat__input--sub"
						type="number"
						value={props.stamina.max}
						onChange={updateMaxStamina}
					/>
					Max
				</label>
			</div>
			<div className="stat luck">
				<label>
					Luck
					<input
						className="stat__input stat__input--main"
						type="number"
						max={props.luck.max}
						value={props.luck.remaining}
						onChange={updateRemainingLuck}
					/>
				</label>
				<label className="sub-label">
					<input
						type="number"
						className="stat__input stat__input--sub"
						value={props.luck.max}
						onChange={updateMaxLuck}
					/>
					Max
				</label>
			</div>
		</div>
	);
}

Stats.defaultProps = {
	skill: 0,
	stamina: { max: 0, remaining: 0 },
	luck: { max: 0, remaining: 0 },
};
