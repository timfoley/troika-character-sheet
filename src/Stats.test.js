import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";

import { Stats } from "./Stats";

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Stats />, div);
	ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
	const { container } = render(<Stats />);
	expect(container.firstChild).toMatchSnapshot();
});
