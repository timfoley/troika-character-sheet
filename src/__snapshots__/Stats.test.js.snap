// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="stats"
>
  <label
    class="stat skill"
  >
    Skill
    <input
      class="skill-stat stat__input stat__input--main"
      type="number"
      value="0"
    />
  </label>
  <div
    class="stat stamina"
  >
    <label>
      Stamina
      <input
        class="stamina-remaining stat__input stat__input--main"
        max="0"
        type="number"
        value="0"
      />
    </label>
    <label
      class="sub-label"
    >
      <input
        class="stamina-max stat__input stat__input--sub"
        type="number"
        value="0"
      />
      Max
    </label>
  </div>
  <div
    class="stat luck"
  >
    <label>
      Luck
      <input
        class="luck-remaining stat__input stat__input--main"
        max="0"
        type="number"
        value="0"
      />
    </label>
    <label
      class="sub-label"
    >
      <input
        class="luck-max stat__input stat__input--sub"
        type="number"
        value="0"
      />
      Max
    </label>
  </div>
</div>
`;
